### PR TITLE
clean engine logs at the end of each test

### PR DESCRIPTION
**Pull request by Ramzeeset**

### CODE REVIEW

File: tests/python/lib/testcase.py [hunk 1]
```diff
@@ -107,6 +107,20 @@ def custom_setup(cls):
...
+    def teardown_method(self, method):
+        self.custom_teardown_method(method)
+
+    @classmethod
+    def custom_setup_method(cls, method):
+        pass
+
+    @classmethod
+    def custom_teardown_method(cls, method):
+        pass
```
Comment (DrDet) [inline tests/python/lib/testcase.py:122]: These methods shouldn't be static
Similar as basic `setup_method` and `teardwon_method`

**Review by DrDet — CHANGES_REQUESTED** [review]

File: tests/python/lib/testcase.py (hunk 1 above)
Comment (Ramzeeset) [reply tests/python/lib/testcase.py]: fix

File: tests/python/lib/testcase.py [hunk 2]
```diff
@@ -217,6 +229,14 @@ def custom_teardown(cls):
...
                 raise RuntimeError("Got unexpected sanitizer log '{}'".format(sanitizer_log))
 
+    @classmethod
+    def custom_setup_method(cls, method):
+        pass
+
+    @classmethod
+    def custom_teardown_method(cls, method):
+        cls.kphp_server._engine_logs = []
+
```
Comment (DrDet) [inline tests/python/lib/testcase.py:239]: and here

Comment (Ramzeeset) [reply tests/python/lib/testcase.py]: fix

**Review by DrDet — APPROVED** [review]

**Pull request merged**